src/causal/base/core.cljc

```diff
@@ -419,17 +419,16 @@
     (deftype CausalBase [cb]
       IPrintWithWriter
       (-pr-writer [o writer opts]
-        (-write writer (str "#causal/base " (pr-str (s/causal->edn o)))))))
+        (-write writer (str "#causal/base " (pr-str cb))))))
 
 #? (:clj (defmethod print-method CausalBase [^CausalBase o ^java.io.Writer w]
-           (.write w (str "#causal/base " (pr-str (s/causal->edn o))))))
+           (.write w (str "#causal/base " (pr-str (.-cb o))))))
 
 (defn read-edn-map
-  [read-object]
-  (let [[cb] read-object]
-    (CausalBase. cb)))
+  [cb]
+  (CausalBase. cb))
 
-#? (:cljs (cljs.reader/register-tag-parser! 'causal.collections.list read-edn-map))
```

Comment:
@smothers This seemed like an error, so I replaced it with a `'causal/base` tag parser.  Will this break anything?

Reply:
This is indeed a bug. Good catch. This code was never tested so it shouldn’t break anything important. Everything important should have tests.